### DISCUSSION TITLE
Get default column value for validation?

### DISCUSSION THREAD

**Discussion by sqcsabbey** [simolus3/drift#2293] [Q&A] [answered]
I have a column with a constant default value, like:

```
  IntColumn get connectRetries => integer().withDefault(const Constant(3))();
```

I'd like to pre-populate this value in a form for inserting a row into this table (without inserting a dummy record and checking its value). 

I tried getting it with `table.connectRetries.defaultValue!.___`, but I didn't know how to get the constant back out of this defaultValue Expression... how would I fill in this blank to get the int `3` here? Or is there another way to get this value?

**Comment by simolus3** [comment, marked as answer]
There is no guarantee that the default value can be represented in Dart. Things like `currentDateAndTime` are also valid constants, and constants defined in drift files are just stored as a `CustomExpression` in generated columns.

That said, if you know that you've used a `Constant` as a default, you can get that back out with `(table.connectRetries.defaultValue as Constant).value`.
